### PR TITLE
Gh free motion adaptive rotation sequence

### PR DESCRIPTION
**Pull request by GerhardHippmann**
This requires consistent update of duplicated code in https://github.com/ModiaSim/Modia.jl/blob/main/src/Modia3D_Interface.jl

### CODE REVIEW

File: src/contactDetection/ContactDetectionMPR/handler.jl [hunk 1]
```diff
@@ -74,14 +74,14 @@ function selectContactPairs!(sim, scene::Composition.Scene, ch::Composition.Cont
         # z[1] ... zero crossing function from contact to no contact
         # max. distance (= min. penetration) of active contact pairs
         if isempty(ch.contactDict)
-            @inbounds simh.z[1] = -42.0
+            simh.z[scene.zStartIndex] = -42.0
```
Comment (AndreaNeumayr) [inline src/contactDetection/ContactDetectionMPR/handler.jl:77]: Why did you remove `@inbounds` ?

File: src/ModiaInterface/model3D.jl [hunk 2]
```diff
@@ -131,18 +131,63 @@ function J132(rot132::AbstractVector)
...
+
+    next_isrot123 = Var(start=true),
+    _rotName = "???",  # is changed by buildModia3D to the full path name of "rot"
+
     equations = :[
-        der(r)   = v
-        der(rot) = J123(rot) * w
+        der(r) = v
+
+        isrot123 = pre(next_isrot123)
```
Comment (AndreaNeumayr) [inline src/ModiaInterface/model3D.jl:184]: What is `pre`?

File: src/Composition/joints/object3DMotion.jl [hunk 3]
```diff
@@ -49,7 +50,8 @@ mutable struct FreeMotion <: Modia3D.AbstractJoint
                           r::AbstractVector   = Modia3D.ZeroVector3D,
                           rot::AbstractVector = Modia3D.ZeroVector3D,
                           v::AbstractVector   = Modia3D.ZeroVector3D,
-                          w::AbstractVector   = Modia3D.ZeroVector3D)
+                          w::AbstractVector   = Modia3D.ZeroVector3D,
+                          next_isrot123::Bool = true)  # dummy argument, is ignored
```
Comment (AndreaNeumayr) [inline src/Composition/joints/object3DMotion.jl:54]: I don't understand `next_isrot123` here. You don't use it in the constructor.

File: src/ModiaInterface/model3D.jl (hunk 2 above)
Comment (MartinOtter) [reply src/ModiaInterface/model3D.jl:184]: > 
> 
> What is `pre`?

Modia language element that corresponds to "pre" in Modelica: During an event iteration, pre(var) is the value of variable "var" from the last event iteration.

Comment (AndreaNeumayr) [reply src/ModiaInterface/model3D.jl:184]: Thank you.

File: src/Composition/joints/object3DMotion.jl (hunk 3 above)
Comment (GerhardHippmann) [reply src/Composition/joints/object3DMotion.jl:54]: Implemented by @MartinOtter.
